server/src/web/ghost/ghost.controller.spec.ts

```diff
@@ -1,20 +1,19 @@
 import { Test, TestingModule } from '@nestjs/testing';
 import { GhostController } from './ghost.controller';
 import {instance, mock, when} from 'ts-mockito';
-import {GhostService} from '../../app/ghost/ghost.service';
 import {Ghost} from '../../domain/ghost/ghost.entity';
+import {GhostServiceImpl} from '../../app/ghost/ghost.service.impl';
 
 describe('Ghost Controller', () => {
-  const mockGhostService: GhostService = mock(GhostService);
+  const mockGhostService: GhostServiceImpl = mock(GhostServiceImpl);
```

Comment:
According to TDD, I recommend mocking interface, because you should write your unit tests before you write your implementation classes.
Also, mocking interface ensures Liskov Substitution Principal (LSP)

Reply:
엇 저부분 왜바깠지 인터페이스가 맞는듯 굿굿 감사!!
수정 후에 다시 요청하겟삼!

Reply:
@als95 
이전에는 ts-mockito가 인터페이스 모킹을 지원하지 않아 구현체를 직접 모킹했던 것 같음! 최근 반영되어, 따라 반영했음.
기존에 ts-mockito에서 interface 모킹을 지원하지 않았다는 게시글: https://github.com/NagRock/ts-mockito/issues/117
최근 인터페이스를 모킹하는 게 가능해졌다는 가이드: https://github.com/NagRock/ts-mockito/releases/tag/v2.4.0

그리고, 디펜던시를 잘 resolve하느냐 테스트하는 부분에서는 여전히 구현체에 의존하고 있는데, 현재 GhostServiceImpl이 빈으로 등록돼있어서인지 인터페이스를 모킹했을 땐 제대로 resolve가 안됨... 그래서 구현체를 모킹했음!

```
describe('dependency resolve', () => {
    it('should be defined', async () => {
      const module: TestingModule = await Test.createTestingModule({
        controllers: [GhostController],
        providers: [{
          provide: 'GhostService',
          useValue: instance(mock(GhostServiceImpl)),
        }],
      }).compile();

      controller = module.get<GhostController>(GhostController);
      expect(controller).toBeDefined();
    });
  });

```